client/src/controllers/TabsController.test.js

```diff
@@ -0,0 +1,370 @@
+import { Application } from '@hotwired/stimulus';
+import { TabsController } from './TabsController';
+
+function simulateKeydown(key) {
+  const event = new KeyboardEvent('keydown', {
+    key,
+    code: key,
+    bubbles: true,
+    cancelable: true,
+  });
+  document.activeElement.dispatchEvent(event);
+}
+
+jest.useFakeTimers();
+
+describe('TabsController', () => {
+  let app;
+  const oldWindowLocation = window.location;
+
+  const setup = async (
+    html = `
+      <div class="w-tabs" data-controller="w-tabs" data-action="popstate@window->w-tabs#loadHistory" data-w-tabs-selected-class="animate-in">
+        <div class="w-tabs__list" role="tablist" data-action="keydown.right->w-tabs#selectNext keydown.left->w-tabs#selectPrevious keydown.home->w-tabs#selectFirst keydown.end->w-tabs#selectLast">
+          <a id="tab-label-tab-1" href="#tab-tab-1" role="tab" tabindex="-1" data-w-tabs-target="label" data-action="w-tabs#select:prevent">
+            Cheese
+          </a>
+          <a id="tab-label-tab-2" href="#tab-tab-2" role="tab" data-action="w-tabs#select:prevent" data-w-tabs-target="label">
+            Chocolate
+          </a>
+          <a id="tab-label-tab-3" href="#tab-tab-3" role="tab" data-action="w-tabs#select:prevent" data-w-tabs-target="label">
+            Coffee
+          </a>
+        </div>
+        <div class="tab-content tab-content--comments-enabled">
```

Comment:
```suggestion
        <div class="tab-content">
```

No need for this class to be added to the test, it could be confusing.